### ISSUE TITLE
Implement cloudstate.crdt.Crdt Service

### ISSUE THREAD

**Issue by marcellanz** [cloudstateio/go-support#17]
Implement the cloudstate.crdt.Crdt Service from protobuf/protocol/cloudstate/crdt.proto

**Tasks**
- [x] CRDT Service
  - [x] State Handling
  - [x] Command Handling
    - [x] Non-Streamed
    - [x] Streamed
  - [x] Cancellation
  - [x] "Callbacks"
  - [x] Context API
- [x] GCounter
- [x] PNCounter
- [x] GSet
- [x] ORSet
- [x] Flag
- [x] LWWRegister
- [x] ORMap
- [x] Vote
- [x] apply new context based API to the evensourced API
- [x] verify implementation (TCK, CrdtSpecs(?))
  - ~Currently blocked by https://github.com/cloudstateio/cloudstate/issues/316, but marking progress on the issue to have a 
TCK capable to verify the CRDT state model.~
  - TCK verification is replaced by functional tests.
- [x] ~~presence example~~
- [x] ~~paradox documentation~~ => see #36 
- [x] ~~Optional: provide CRDT protocol spec feedback (mainly for clarification without reading the implementation)~~

**WIP-branch**
https://github.com/cloudstateio/go-support/tree/feature/crdt_support

**Comment by marcellanz** [comment]
Working on it.

**Comment by viktorklang** [comment]
@marcellanz Very cool, Marcel! (Ping @jroper)

**Comment by marcellanz** [comment]
- [ ] CRDT Service
- [ ] presence example
- [ ] GCounter
- [ ] PNCounter
- [ ] GSet
- [ ] ORSet
- [ ] Flag
- [ ] LWWRegister
- [ ] ORMap
- [ ] Vote

**Comment by viktorklang** [comment]
Woot! *very* nice progress here, @marcellanz!

**Comment by sleipnir** [comment]
Awesome @marcellanz

**Comment by marcellanz** [comment]
Input from #27:
>in future, we may add features to Cloudstate where we start up additional versions of the same entity, for example, we might offer some built in temporal loading support, where users can say "run this command against this entity in the state that it was in at this point of time in history". To do that, we would create a new entity context for that, even if an existing one already existed on that node, and replay the journal up until that time before running the command.

The [current CRDT `runner`](https://github.com/cloudstateio/go-support/blob/feature/crdt_support/cloudstate/crdt/server.go#L99) should do the same for EventSourcing and any other state model.

**Comment by marcellanz** [comment]
TCK verification is replaced by functional tests.

**Comment by marcellanz** [comment]
"Optional: provide CRDT protocol spec feedback (mainly for clarification without reading the implementation)" is captured in a separate issue: https://github.com/cloudstateio/go-support/issues/40
